compiler/rustc_middle/src/mir/interpret/mod.rs

```diff
@@ -588,3 +588,20 @@ pub fn read_target_uint(endianness: Endian, mut source: &[u8]) -> Result<u128, i
     debug_assert!(source.len() == 0); // We should have consumed the source buffer.
     uint
 }
+
+/// Read a pointer-sized thing from a target, so it will fit in a u64
+/// as the modules using this don't support wider pointer values
+/// FIXME(jubilee): Move this out of here and closer to the things using it!
+#[inline]
+pub fn read_target_ptr(endian: Endian, ptr_size: usize, bytes: &[u8]) -> u64 {
```

Comment:
This should probably take `Size` instead of `usize`.

Reply:
Size is a u64, so on a 32-bit host that is over-weighting this input (even tho' the output should be u64 regardless).

I thought about adding an enum to rustc_target so that can be matched on directly for more type-strictness, though.

Reply:
It's just standard to use `Size` to describe such a property of the target, and the consistency helps avoid bugs. Also 32-bit host *performance* shouldn't be a significant concern (even tho memory usage in data structures may be).

Reply:
@workingjubilee, this code doesn't seem terribly idiomatic, and if your concern is input overhead, why not just call `bytes.len()`? Besides, it's definitely going to be optimized away during inlining, and if you call `bytes.len()` then it's easier for the optimizer to see what the variable means.

Reply:
@workingjubilee Oh, you're probably right! Even taking an input seems a bit silly, now that I think of it. I must be reading too many C codebases lately. 😅